### PR TITLE
[ioctl] support querying delegate by operator address

### LINKED ISSUE #3900: [ioctl] bc delegate command to accept operation address
### Description of new feature request
an `ioctl bc delegate` command is newly added, like below
```
ioctl bc delegate cpc --endpoint=api.iotex.one:443
{
	name: cpc
	ownerAddress: io1smh0a4ms9fc5p8v9h9l7ffgkekv032hzsdkr9a
	operatorAddress: io16fknjsnhj6m5x2fawu5tqfrk6pr9r9xyz489vk
	rewardAddress: io12mgttmfa2ffn9uqvn0yn37f4nz43d248l2ga85
	totalWeightedVotes: 68963450.120222472655746524
	selfStakeBucketIdx: 12
	selfStakingTokens: 2975349.881825114451272576 IOTX
}
```
make this command to also accept the operator address as input, that is 
```
ioctl bc delegate io16fknjsnhj6m5x2fawu5tqfrk6pr9r9xyz489vk --endpoint=api.iotex.one:443
```
will output the same result

### Is this a hard-fork (and height to be activated)
- [] Yes
- Name of activation height:

### Implementation plan
- [] Phase 1 description (PR number)
- [] Phase 2 description (PR number)
- [] Phase 3 description (PR number)

### Affected modules
- [] action
- [] actpool
- [] api
- [] blockchain
- [] blockindex
- [] blocksync
- [] chainservice
- [] config
- [] consensus
- [] db
- [] dispatcher
- [x] ioctl
- [] p2p
- [] server
- [] state
- [] tools

### PR DESCRIPTION
**Pull request by envestcc**
# Description

Fixes #3900 

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: ioctl/cmd/bc/bcdelegate.go [hunk 1]
```diff
@@ -142,3 +150,77 @@ func getDelegateByName(name string) (*iotextypes.CandidateV2, error) {
...
+			return candidateList.Candidates[idx], nil
+		}
+		if len(candidateList.Candidates) < readCandidatesLimit {
+			break
+		}
+	}
+	return nil, output.NewError(output.UndefinedError, "failed to find delegate", nil)
+}
+
+func getDelegates(cli iotexapi.APIServiceClient, offset, limit uint32) (candidateList *iotextypes.CandidateListV2, err error) {
```
Comment (dustinxie) [inline ioctl/cmd/bc/bcdelegate.go:182]: there is already `func getStakingCandidates()` in `nodedelegate.go`
move it to /util so can re-use the code?

File: ioctl/cmd/bc/bcdelegate.go [hunk 2]
```diff
@@ -142,3 +150,77 @@ func getDelegateByName(name string) (*iotextypes.CandidateV2, error) {
...
+
+func getDelegateByAddress(addr string) (*iotextypes.CandidateV2, error) {
+	conn, err := util.ConnectToEndpoint(config.ReadConfig.SecureConnect && !config.Insecure)
+	if err != nil {
+		return nil, output.NewError(output.NetworkError, "failed to connect to endpoint", err)
+	}
+	defer conn.Close()
+	cli := iotexapi.NewAPIServiceClient(conn)
+
+	readCandidatesLimit := 20000
```
Comment (dustinxie) [inline ioctl/cmd/bc/bcdelegate.go:162]: 200 works and is maybe better?

File: ioctl/cmd/bc/bcdelegate.go (hunk 1 above)
Comment (envestcc) [reply ioctl/cmd/bc/bcdelegate.go]: good advice

**Review by dustinxie — APPROVED** [review]

File: ioctl/util/chain_api.go [hunk 3]
```diff
@@ -13,6 +13,7 @@ const (
 	_stakingProtocol = "staking"
 )
 
+// GetStakingBuckets returns staking buckets
```
Comment (dustinxie) [inline ioctl/util/chain_api.go:16]: GetStakingCandidates returns ...

**Review by Liuhaai — APPROVED** [review]

**Pull request merged**